Formalizitons of four theorems: Denumerability_of_the_Rational_Number…

ngongatlu opened the pull request:
…s, Isosceles_Triangle_Theorem_(Pons_asinorum),  Law_of_Cosines, Fundamental_Theorem_of_Arithmetic

ngongatlu committed:
Formalizitons of four theorems: Denumerability_of_the_Rational_Numbers, Isosceles_Triangle_Theorem_(Pons_asinorum),  Law_of_Cosines, Fundamental_Theorem_of_Arithmetic

ngongatlu committed:
Formalizations of 04 theorems: Leibniz's_Series_for_Pi, Non-Denumerability_of_the_Continuum, Product_of_Segments_of_Chords, Ptolemy's_Theorem

ngongatlu committed:
Formalization of 04 theorems: Sum_of_the_Angles_of_a_Triangle, Cauchy's_theorem_(group_theory), Order_of_a_Subgroup, Sylow's_Theorem

ngongatlu committed:
Formalizations of 06 theorem: Green-Tao_theorem, Perfect_Number_Theorem, Birthday_Problem, Triangle_Inequality, Derangements_Formula, Principle_of_Inclusion-Exclusion

ngongatlu commented:
Dear Prof. Tom,
We have just sent pull request for 18 theorems. We will send you formalizations of 7 theorems in next time.
Best,
TLU group.

ngongatlu committed:
Edit file Cauchy's theorem (group theory).rst

We send you the new file replace Cauchy's theorem (group theory).rst file

ngongatlu committed:
Edit file Order of a Subgroup.rst

replace by new file

ngongatlu committed:
edit file Sylow's_Theorem.rst

replace by better file

thalesant reviewed (changes requested):
Requested changes.

Fix Cauchy's theorem (it is the not the same as Lagrange).

3 universes, alpha, beta are not needed in group theory articles.

In the law of cosines, only one of the three statements is needed, because the others merely rename variables.

The product over a list is already defined in [Lean lists prod](https://github.com/leanprover-community/mathlib/blob/master/src/data/list/defs.lean). There is no need to redefine it.  Also look at how it is defined using foldl, and try to imitate that style in your other folding operations such as sum.

We should also explore if there is a better way to express problems involving finset and set such as inclusion-exclusion and the birthday paradox.  Try using type class resolution, using the instances provided in [finset Lean file](https://github.com/leanprover-community/mathlib/blob/master/src/data/finset.lean).  For example, union is already defined on finset (line 259)[https://github.com/leanprover-community/mathlib/blob/master/src/data/finset.lean].  Can you use the has_lift instance (line 51) to do away with the explicit conversions from finset to set?

ngongatlu committed:
Solution_of_a_Cubic.rst	 Heron's_Formula.rst	 Solution_of_the_General_Quartic_Equation.rst	Formalizations of 07 next theorems:  Theorem_of_Ceva, Cauchy-Schwarz_Inequality, Formula_for_the_Number_of_Combinations, Desargues's_Theorem

These are 07 formalizations of next theorems.

ngongatlu committed:
Edit 05 theorems base on Prof. Tom's comment

Birthday_Problem, Principle_of_Inclusion-Exclusion, Derangements_Formula: use fintype to take cardinality. 
Law_of_Cosines: delete two last propositions in the statement of the theorem.
Fundamental_Theorem_of_Arithmetic: use function 'prod' to calculate production of all elements of a list.

ngongatlu committed:
edit some theorems

edit some mistakes

thalesant reviewed (changes requested):
Excellent progress.  Here is a review.

Desarges's theorem might need nondegeneracy conditions a \ne b.

Fermat must assume that the natural numbers are nonzero.

Heron's theorem cannot be stated until we have the 2-dimensional Lebesgue measure.  We need to wait to do this formal abstract.  The area of a circle has the same issue.

Or should be changed to and in the cubic and quartic equations.

In the cubic problem and quartic equations, it is not clear what the exponent (-)^(1/2) and (-)^(1/3) mean, because extracting roots is a multi-valued function on the field of complex numbers.

In the quartic problem, it is not clear what the syntax "let is_root (cubic A B C D) y in..." means.  The usual syntax is "let x := A in ..."
